src/main/kotlin/no/nav/familie/ba/sak/kjerne/beregning/TilkjentYtelseGenerator.kt

```diff
@@ -0,0 +1,139 @@
+package no.nav.familie.ba.sak.kjerne.beregning
+
+import no.nav.familie.ba.sak.common.Feil
+import no.nav.familie.ba.sak.kjerne.beregning.AndelTilkjentYtelseGenerator.oppdaterAndelerMedEndretUtbetalingAndeler
+import no.nav.familie.ba.sak.kjerne.beregning.TilkjentYtelseUtils.oppdaterTilkjentYtelseMedEndretUtbetalingAndelerGammel
+import no.nav.familie.ba.sak.kjerne.beregning.UtvidetBarnetrygdUtil.finnUtvidetVilkår
+import no.nav.familie.ba.sak.kjerne.beregning.domene.AndelTilkjentYtelse
+import no.nav.familie.ba.sak.kjerne.beregning.domene.AndelTilkjentYtelseMedEndreteUtbetalinger
+import no.nav.familie.ba.sak.kjerne.beregning.domene.EndretUtbetalingAndelMedAndelerTilkjentYtelse
+import no.nav.familie.ba.sak.kjerne.beregning.domene.InternPeriodeOvergangsstønad
+import no.nav.familie.ba.sak.kjerne.beregning.domene.TilkjentYtelse
+import no.nav.familie.ba.sak.kjerne.beregning.domene.YtelseType
+import no.nav.familie.ba.sak.kjerne.endretutbetaling.domene.Årsak
+import no.nav.familie.ba.sak.kjerne.fagsak.FagsakType
+import no.nav.familie.ba.sak.kjerne.grunnlag.personopplysninger.PersonType
+import no.nav.familie.ba.sak.kjerne.grunnlag.personopplysninger.PersonopplysningGrunnlag
+import no.nav.familie.ba.sak.kjerne.personident.Aktør
+import no.nav.familie.ba.sak.kjerne.vilkårsvurdering.domene.Vilkårsvurdering
+import java.time.LocalDate
+
+object TilkjentYtelseGenerator {
+    fun beregnTilkjentYtelse(
+        vilkårsvurdering: Vilkårsvurdering,
+        personopplysningGrunnlag: PersonopplysningGrunnlag,
+        endretUtbetalingAndeler: List<EndretUtbetalingAndelMedAndelerTilkjentYtelse> = emptyList(),
+        fagsakType: FagsakType,
+        skalBrukeNyVersjonAvOppdaterAndelerMedEndringer: Boolean = true,
+        hentPerioderMedFullOvergangsstønad: (aktør: Aktør) -> List<InternPeriodeOvergangsstønad> = { _ -> emptyList() },
+    ): TilkjentYtelse {
+        val tilkjentYtelse =
+            TilkjentYtelse(
+                behandling = vilkårsvurdering.behandling,
+                opprettetDato = LocalDate.now(),
+                endretDato = LocalDate.now(),
+            )
+
+        val (endretUtbetalingAndelerSøker, endretUtbetalingAndelerBarna) = endretUtbetalingAndeler.partition { it.person?.type == PersonType.SØKER }
+
+        val andelerTilkjentYtelseBarnaUtenEndringer =
+            OrdinærBarnetrygdUtil
+                .beregnAndelerTilkjentYtelseForBarna(
+                    personopplysningGrunnlag = personopplysningGrunnlag,
+                    personResultater = vilkårsvurdering.personResultater,
+                    fagsakType = fagsakType,
+                ).map {
+                    if (it.person.type != PersonType.BARN) throw Feil("Prøver å generere ordinær andel for person av typen ${it.person.type}")
```

Comment:
Er dette mer forklarende kanskje? 

```suggestion
                    if (it.person.type != PersonType.BARN) throw Feil("Prøver å generere ordinær andel for person av typen ${it.person.type}. Forventet ${PersonType.BARN}")
```

Reply:
Det ligger litt indirekte i at man prøver å generere ordinær andel (som kun kan være på barnet), men gjør ikke noe å legge det til 🙌